GetMissingEntries validation fail when CRD object has float64 data types

`float64` data types not supported when running `Schema.GetMissingEntries` method.
Besides, returned `SchemaEntry.Path` is empty.

Steps to reproduce
===============

*CRD file
```yaml
apiVersion: apiextensions.k8s.io/v1beta1                       
kind: CustomResourceDefinition                                 
metadata:                                                      
  name: plans.capabilities.3scale.net                          
spec:                                                          
  group: capabilities.3scale.net                               
  names:                                                       
    kind: Plan                                                 
    listKind: PlanList                                         
    plural: plans                                              
    singular: plan                                             
  scope: Namespaced                                            
  subresources:                                                
    status: {}                                                 
  validation:                                                  
    openAPIV3Schema:                                           
      properties:                                              
        spec:                                                  
          type: object                                         
          properties:                                          
            costMonth:                                         
              format: double                                   
              type: number                                     
        status:                                                
          type: object                                         
  version: v1alpha1                                            
  versions:                                                    
  - name: v1alpha1                                             
    served: true                                               
    storage: true                           
```


Validation test:

```go
package main

import (
	"fmt"
	"io/ioutil"

	"github.com/RHsyseng/operator-utils/pkg/validation"
)

type PlanSpec struct {
	CostMonth float64 `json:"costMonth,omitempty"`
}

type PlanStatus struct {
}

type Plan struct {
	Spec   PlanSpec   `json:"spec,omitempty"`
	Status PlanStatus `json:"status,omitempty"`
}

func getSchema(crd string) validation.Schema {
	bytes, err := ioutil.ReadFile(crd)
	if err != nil {
		panic(fmt.Sprintf("Error reading CRD yaml from %v", crd))
	}
	schema, err := validation.New(bytes)
	if err != nil {
		panic(err)
	}
	return schema
}

func main() {
	schemaFile := "plan_crd.yaml"
	schema := getSchema(schemaFile)
	missingEntries := schema.GetMissingEntries(&Plan{})
	for _, missing := range missingEntries {
		panic(fmt.Sprintf("Missing or incorrect schema validation at %s, expected type %s", missing.Path, missing.Type))
	}

	fmt.Println("OK")
}
```
After compiling, running the Go program outputs:
```shell
$ ./validation 
panic: Missing or incorrect schema validation at /spec/costMonth, expected type
```

--- a/pkg/validation/schema_sync.go
+++ b/pkg/validation/schema_sync.go
@@ -91,6 +91,10 @@ func equivalentSchemaType(kind reflect.Kind) string {
 	switch kind {
 	case reflect.String:
 		return "string"
+	case reflect.Float32:
+		return "number"
+	case reflect.Float64:
+		return "number"
 	case reflect.Int:
 		return "integer"
 	case reflect.Int8:

--- a/pkg/validation/schema_sync_test.go
+++ b/pkg/validation/schema_sync_test.go
@@ -86,6 +86,55 @@ func TestSchemaStructSlice(t *testing.T) {
 	}
 }
 
+func TestSchemaFloat64(t *testing.T) {
+	schemaYaml := `
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: sample.app.example.com
+spec:
+  group: app.example.com
+  names:
+    kind: SampleApp
+    listKind: SampleAppList
+    plural: sampleapps
+    singular: sampleapp
+  scope: Namespaced
+  version: v1
+  validation:
+    openAPIV3Schema:
+      required:
+        - spec
+      properties:
+        spec:
+          type: object
+          required:
+          - number
+          properties:
+            number:
+              type: number
+              format: double
+`
+	schema, err := New([]byte(schemaYaml))
+	assert.NoError(t, err)
+
+	type myAppSpec struct {
+		Number float64 `json:"number,omitempty"`
+	}
+
+	type myApp struct {
+		Spec   myAppSpec       `json:"spec,omitempty"`
+	}
+
+	cr := myApp{
+		Spec: myAppSpec{
+			Number: float64(23),
+		},
+	}
+	missingEntries := schema.GetMissingEntries(&cr)
+	assert.Len(t, missingEntries, 0, "Expect no missing entries in CRD for this struct: %v", missingEntries)
+}
+
 func getCompleteSchema(t *testing.T) Schema {
 	schemaYaml := `
 apiVersion: apiextensions.k8s.io/v1beta1

